### PR TITLE
Bind all methods of Auth0VueClient to this

### PR DESCRIPTION
**Pull request by frederikprijck**
There seems to be an issue when setting `this` as the value when calling Vue's `provide`. To solve it, we should bind all methods using `this`.
[Vuex appeared to have solved it in the same way](https://github.com/vuejs/vuex/blob/main/src/store.js#L50).

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -48,7 +48,24 @@ export class Auth0Plugin implements Auth0VueClient {
...
+    this.logout = this.logout.bind(plugin);
+    this.getAccessTokenSilently = this.getAccessTokenSilently.bind(plugin);
+    this.getAccessTokenWithPopup = this.getAccessTokenWithPopup.bind(plugin);
+    this.checkSession = this.checkSession.bind(plugin);
+    this.handleRedirectCallback = this.handleRedirectCallback.bind(plugin);
+    this.buildAuthorizeUrl = this.buildAuthorizeUrl.bind(plugin);
+    this.buildLogoutUrl = this.buildLogoutUrl.bind(plugin);
+    this.__checkSession = this.__checkSession.bind(plugin);
+    this.__refreshState = this.__refreshState.bind(plugin);
+    this.__proxy = this.__proxy.bind(plugin);
```
Comment (stevehobbsdev) [inline src/plugin.ts:67]: Just a question, is there a dynamic way we can do this (e.g. find all functions that are direct members of `this` and bind it)? I can just see this being missed later; if we add new methods later we're likely to forget to add it to this list and run into issues when someone tries to use it.

Comment (frederikprijck) [reply src/plugin.ts]: Updated the PR to do it more dynamic.

File: src/plugin.ts [hunk 2]
```diff
@@ -148,7 +153,7 @@ export class Auth0Plugin implements Auth0VueClient {
     }
   }
 
-  async __refreshState() {
+  private async __refreshState() {
```
Comment (frederikprijck) [inline src/plugin.ts:156]: Made this private as it shouldnt be public

File: src/plugin.ts (hunk 1 above)
Comment (stevehobbsdev) [reply src/plugin.ts]: Very cool

**Review by stevehobbsdev — APPROVED** [review]
Looks good, although Browserstack tests are failing.

Also not sure about the FOSSA check though.

**Pull request merged**